contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql

```diff
@@ -4267,3 +4267,16 @@ END
 $$;
 GRANT EXECUTE ON PROCEDURE sys.sp_changedbowner(IN sys.sysname, IN sys.VARCHAR(5)) TO PUBLIC;
 
+-- Revoke public execute for specific procedures
+REVOKE EXECUTE ON PROCEDURE sys.babel_datatype_precedence_initializer FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.babel_coercion_initializer FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.babel_catalog_initializer FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.babel_create_builtin_dbs FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.babel_drop_all_dbs FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.babel_initialize_logins FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.babel_drop_all_logins FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.initialize_babel_extras FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.analyze_babelfish_catalogs FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.initialize_babelfish FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.remove_babelfish FROM PUBLIC;
```

Comment:
After this change only the function owner will have execute rights on it, right? Just thinking if we can add positive test case for them somehow.

Reply:
Otherwise changes looks good.

Reply:
Yes, will test it locally.
But all these procedures are executed while installing or removing babelfish (directly or indirectly). That could be considered as a tests. Right ?

Reply:
Yes.

Reply:
we should add tests where 1) Existing login/user which can execute these procedures 2) Create a new login/user which can execute these procedures 3) Corresponding PSQL behavior 4) Additional tests to cover Gitfarm side of differences